### ISSUE TITLE
Button warning and warning solid variants as only colour difference?

### ISSUE THREAD

**Issue by microbit-grace** [microbit-foundation/ui#21]
Similar to [Button toolbar variants only varies the focus ring colour issue](https://redirect.github.com/microbit-foundation/ui/issues/20), I wonder if the warning variants should be variants. For example, in Chakra, they seem to not count it as a separate variant.

A warning solid button is pretty much a button with "brand" colour replaced with "danger".

But for a warning button, there is a bit more difference when compared with a secondary button (see numbers for color and _active borderColor):

  secondary: {
    borderWidth: "2px",
    borderColor: "brand.500",
    color: "brand.**700**",
    bg: "transparent",
    _hover: { borderColor: "brand.600" },
    _active: { bg: "brand.50", borderColor: "brand.**700**" },
  },

 warning: {
        borderWidth: "2px",
        borderColor: "danger.500",
        color: "danger.**500**",
        bg: "transparent",
        _hover: { borderColor: "danger.600" },
        _active: { bg: "danger.50", borderColor: "danger.**500**" },
      },

**Current warning button**
<img width="111" height="56" alt="Image" src="https://github.com/user-attachments/assets/ef92a7ec-fcb7-414b-9696-afbd16a9981b" />

**Secondary button with danger color**
<img width="98" height="44" alt="Image" src="https://github.com/user-attachments/assets/9fd59be7-d578-48f5-b988-6ba9c8a7b9c4" />

We tend to use a darker colour text colour in secondary buttons to satisfy text colour contrast. The current warning button doesn't do that. This might explain why the current warning button text appear to not have sufficient text colour contrast (potentially a separate issue), or perhaps it is to do with colour selection (also potentially a separate issue).

**Comment by microbit-matt-hillsdon** [comment]
I think this is in a better place now with tone="danger". We still have solid/outline and primary/secondary because classroom/data use a black button theme for primary/secondary which isn't just a ramp swap. But you can still reach for solid/outline tone=danger when needed in those apps.
